src/mixer/basetrackplayer.h

```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include "control/controlencoder.h"
```

Comment:
yes, but nope, here the forward declaration is sufficient, otherwise all units that include this header also (unnecessarily) include controlencoder.h
That goes to basetrackplayer.cpp

Reply:
basetrackplayer.cpp
(not controlpickermenu.cpp, that is just a description / mapping utility, it doesn't create or touch controls)

Reply:
It was too late to work '^^
Fidex, hope it compiles